dkist/dataset/tiled_dataset.py

```diff
@@ -150,6 +128,27 @@ def __init__(
         self._data = np.ma.masked_array(dataset_array, dtype=object, mask=mask)
         meta = meta or {}
         inventory = meta.get("inventory", inventory or {})
+
+        # If headers are saved as one Table for the whole TiledDataset, use those first
+        # Otherwise stack the headers saved for component Datasets
+        if meta.get("headers") is None:
+            meta["headers"] = vstack(
+                [Table(ds.headers) if ds else Table({}) for ds in self._data.compressed()]
+            )
+        # Then distribute headers (back) out to component Datasets as slices of the main Table
+        # for row in dataset_array:
+        #     for ds in row:
+
+        offset = 0
```

Comment:
I think we should only be doing this split out if `meta.get("headers") is None`, i.e if we haven't loaded from asdf.

Reply:
If we don't then the tile headers and the combined_headers aren't sharing memory because vstack creates a copy.

Reply:
what I mean is we should only do this next bit if we have vstacked not if we already had combined headers (in which case we should assume they are correct).